packages/fuels-abigen-macro/tests/harness.rs

```diff
@@ -1085,6 +1058,9 @@ async fn test_provider_launch_and_connect() {
         .unwrap();
     assert_eq!(42, result.value);
 
+    wallet.set_provider(launched_provider);
```

Comment:
I'm a bit confused about this test case now that we've made these changes. It feels like there's no difference between what all the other tests are doing; deploying is a different process than connecting to a deployed contract. Every test in here deploys and connects, and I'm not sure about the value of changing the provider here (`wallet.set_provider()`). Is this test case (`test_provider_launch_and_connect`) completely redundant now?

Reply:
My understanding is that the test checks if behavior is consistent when making calls on a launched vs. connected `provider` instances. The binding now uses the `provider` from the supplied `wallet` to make method calls so I change it to `launched_provider` before making the second call.

Reply:
Fair enough, not entirely redundant then!

Reply:
If this is the main test for testing `connected` vs `launched` instances, then the test `connect_to_deployed_contract` can be removed because it does this (actually badly now haha)